dht.go

```diff
@@ -737,12 +737,11 @@ func (dht *IpfsDHT) FindLocal(ctx context.Context, id peer.ID) peer.AddrInfo {
 	_, span := internal.StartSpan(ctx, "IpfsDHT.FindLocal", trace.WithAttributes(attribute.Stringer("PeerID", id)))
 	defer span.End()
 
-	switch dht.host.Network().Connectedness(id) {
-	case network.Connected, network.CanConnect:
+	connectedness := dht.host.Network().Connectedness(id)
+	if connectedness == network.Connected || connectedness == network.Limited {
```

Comment:
nit: could we have a wraper so we dont have the same boilerplate everywhere?

```go
func hasValidConnectdness(host, id) bool {
  connectedness := host.Network().Connectedness(id)
	return connectedness == network.Connected || connectedness == network.Limited {
}
```

Reply:
We might want to do that. I just wanna make sure that we do indeed want to this check everywhere I did in this PR.